pixeltable/type_system.py

```diff
@@ -918,6 +915,36 @@ def matches(self, other: ColumnType) -> bool:
     def __hash__(self) -> int:
         return hash((self._type, self.nullable, self.shape, self.dtype))
 
+    def is_supertype_of(self, other: ColumnType, ignore_nullable: bool = False) -> bool:
```

Comment:
It shouldn't be necessary to implement this directly (if the default implementation isn't working, it suggests a bug in `supertype()` )